include/libsemigroups/presentation.tpp

```diff
@@ -185,6 +185,33 @@ namespace libsemigroups {
     return result;
   }
 
+  template <typename Word>
+  void Presentation<Word>::remove_generator_no_checks(
+      typename Presentation<Word>::letter_type x) {
+    size_t index = _alphabet_map[x];
+    _alphabet_map.erase(x);
+    typename Word::iterator start = _alphabet.begin() + index;
+    for (auto it = start + 1; it != _alphabet.end(); ++it) {
+      --_alphabet_map[*it];
+    }
+    _alphabet.erase(start, start + 1);
+  }
+
+  // TODO(now) should this check that the alphabet is valid, as strange things
+  // happen if there you try to remove a duplicate letter from the alphabet?
+  template <typename Word>
+  void Presentation<Word>::remove_generator(
+      typename Presentation<Word>::letter_type x) {
+    if (in_alphabet(x)) {
+      remove_generator_no_checks(x);
+    } else {
+      LIBSEMIGROUPS_EXCEPTION("the argument {} does not belong to the alphabet "
+                              "{}, expected an existing letter",
+                              detail::to_printable(x),
+                              detail::to_printable(alphabet()));
+    }
+  }
```

Comment:
What should the checks for this function be? The current option is the bare minimum; check `x` is in the alphabet. However, if `x` is duplicated in the alphabet, then strange things can happen. I wasn't sure where to place this on the "let the user do what they want" to "don't let the user break things" scale.

Reply:
Is this still an issue?

Reply:
@james-d-mitchell yes, still not sure what the best course of action is. I think the most sensible option is to check that the alphabet is valid; the user should only be able to significantly mess up the internals of a presentation through a `_no_checks` function. What do you think?